mods/persistence/controllers/subsystems/persistence/persistence_loading.dm

```diff
@@ -0,0 +1,297 @@
+///a player enters limbo.
+/datum/controller/subsystem/persistence/proc/_update_limbo_state()
+	// TODO: Generalize this for other things in limbo.
+	for(var/datum/mind/char_mind in global.player_minds)
+		try
+			one_off.RemoveFromLimbo(char_mind.unique_id, LIMBO_MIND)
+		catch(var/exception/e)
+			_handle_recoverable_load_exception(e, "while updating off-world storage state for player '[char_mind.key]'")
+
+///Deserialize cached turfs exclusively from the db cache.
+/datum/controller/subsystem/persistence/proc/_deserialize_turfs()
+	var/list/turfs_loaded = list()
+	var/time_start        = REALTIMEOFDAY
+
+	report_progress_serializer("Deserializing [LAZYLEN(serializer.resolver.things)] cached atoms...")
+	sleep(5)
+
+	for(var/TKEY in serializer.resolver.things)
+		var/datum/persistence/load_cache/thing/T
+		try
+			T = serializer.resolver.things[TKEY]
+			if(ispath(T.thing_type, /datum/wrapper_holder)) // Special handling for wrapper holders since they don't have another reference.
+				serializer.DeserializeDatum(T)
+				continue
+			if(!T.x || !T.y || !T.z)
+				continue // This isn't a turf or a wrapper holder. We can skip it.
+			serializer.DeserializeDatum(T)
+			turfs_loaded["([T.x], [T.y], [T.z])"] = TRUE
+		catch(var/exception/E)
+			to_world_log("Failed to load turf '[T]'!: [EXCEPTION_TEXT(E)]")
+		CHECK_TICK
+
+	in_loaded_world = LAZYLEN(turfs_loaded) > 0
+	. = turfs_loaded
+	report_progress_serializer("Deserialized [LAZYLEN(turfs_loaded)] turfs and their contents in [REALTIMEOFDAY2SEC(time_start)]s.")
+	sleep(5)
+
+/// TODO
+/datum/controller/subsystem/persistence/proc/_setup_default_turfs(var/list/turfs_loaded)
+	var/time_start = REALTIMEOFDAY
+	for(var/datum/persistence/load_cache/z_level/z_level in serializer.resolver.z_levels)
+		var/change_turf = z_level.default_turf && !ispath(z_level.default_turf, /turf/space)
+
+		// Create the areas in the z-level if they don't already exist.
+		// Areas are added to the area dictionary in area/New()
+		for(var/list/area_chunk in z_level.areas)
+			var/area/area_instance = global.area_dictionary["[area_chunk[1]], [area_chunk[2]]"]
+			if(!area_instance)
+				var/new_type = text2path(area_chunk[1])
+				new new_type(null, area_chunk[2])
+		// The areas are split into horizontal chunks with the area type and name corresponding to a certain amount of tiles in a row.
+		var/chunk_index = 1
+		var/list/current_area_chunk
+		var/area/current_area
+		var/turf_count = 1
+		if(length(z_level.areas))
+			current_area_chunk = z_level.areas[chunk_index]
+			current_area = global.area_dictionary["[current_area_chunk[1]], [current_area_chunk[2]]"]
+
+		for(var/turf/T in block(locate(1, 1, z_level.new_index), locate(world.maxx, world.maxy, z_level.new_index)))
+			try
+				if(current_area)
+					current_area.contents += T //#FIXME: It's dangerous to do it like this. Use loc on the turf, not the area's contents "list".
```

Comment:
You can make this change now if you'd like.

Reply:
I'm just leaving comments for later. This isn't the PR to make changes like that.